src/ghdcbot/adapters/storage/sqlite.py

```
@@ -0,0 +1,708 @@
+        """Check if notification was already sent (deduplication)."""
+        with self._connect() as conn:
+            row = conn.execute(
+                "SELECT 1 FROM notifications_sent WHERE dedupe_key = ?",
+                (dedupe_key,),
+            ).fetchone()
+        return row is not None
+
+    def mark_notification_sent(
+        self,
+        dedupe_key: str,
+        event: Any,
```

Ruff (0.14.14)

[error] Undefined name `Any`

(F821)